### PR TITLE
[EC-156] Remove erasing of nodes in MPT

### PR DESCRIPTION
**Pull request by AlanVerbner**
## Description

This commit removes old mpt nodes deletion from underlying db. From now on, a pruning system needs to be performed in order to free space.

## Implementation note

Mpt now doesn't _know_ that some nodes are useless now. If we think our pruning implementation requires to record "deletes" - for example if it's made using reference counting method - this commit can be rolled back.

### CODE REVIEW

File: src/it/scala/io/iohk/ethereum/mpt/MerklePatriciaTreeIntegrationSuite.scala [hunk 1]
```diff
@@ -1,11 +1,13 @@
 package io.iohk.ethereum.mpt
```
Comment (AlanVerbner) [inline src/it/scala/io/iohk/ethereum/mpt/MerklePatriciaTreeIntegrationSuite.scala]: Sorry for the formatting changes on this file

Comment (adamsmo) [inline src/it/scala/io/iohk/ethereum/mpt/MerklePatriciaTreeIntegrationSuite.scala]: maybe use diff to revert the changes, I mean formatting in places that you did not change code?

**Comment by adamsmo** [comment]
besides formatting LGTM

File: src/it/scala/io/iohk/ethereum/mpt/MerklePatriciaTreeIntegrationSuite.scala (hunk 1 above)
Comment (rtkaczyk) [inline src/it/scala/io/iohk/ethereum/mpt/MerklePatriciaTreeIntegrationSuite.scala]: These formatting changes are the result of functional changes in the test code and they are required. He just feels sorry it makes a bit harder to review.
